com.unity.render-pipelines.core/Editor/Settings/RenderPipelineGlobalSettingsProvider.cs

```diff
@@ -0,0 +1,154 @@
+            base.OnActivate(searchContext, rootElement);
+        }
+
+        /// <summary>
+        /// This method is being called when the provider is deactivated
+        /// </summary>
+        public override void OnDeactivate()
+        {
+            DestroyEditor();
+            base.OnDeactivate();
+        }
+
+        /// <summary>
+        /// Ensures that the <see cref="RenderPipelineGlobalSettings"/> asset is correct
+        /// </summary>
+        protected abstract void Ensure();
+
+        /// <summary>
+        /// Creates a new <see cref="RenderPipelineGlobalSettings"/> asset
+        /// </summary>
+        protected abstract void Create(bool useProjectSettingsFolder, bool activateAsset);
+
+        /// <summary>
+        /// Clones the <see cref="RenderPipelineGlobalSettings"/> asset
+        /// </summary>
+        protected abstract void Clone(RenderPipelineGlobalSettings src, bool activateAsset);
+
+        /// <summary>
+        /// Method called to render the IMGUI of the settings provider
+        /// </summary>
+        /// <param name="searchContext">The search content</param>
+        public override void OnGUI(string searchContext)
+        {
+            using (new SettingsProviderGUIScope())
+            {
+                if (renderPipelineSettings == null)
+                {
+                    CoreEditorUtils.DrawFixMeBox(string.Format(Styles.warningGlobalSettingsMissing, ObjectNames.NicifyVariableName(typeof(TGlobalSettings).Name)), () => Ensure());
+                }
+                else
+                {
+                    DrawAssetSelection();
+
+                    if (RenderPipelineManager.currentPipeline != null && !(RenderPipelineManager.currentPipeline is TRenderPipeline))
+                    {
+                        EditorGUILayout.HelpBox(string.Format(Styles.warningSRPNotActive, ObjectNames.NicifyVariableName(RenderPipelineManager.currentPipeline.GetType().Name)), MessageType.Warning);
+                    }
+
+                    if (m_Editor == null)
+                        m_Editor = Editor.CreateEditor(renderPipelineSettings);
+
+                    m_Editor.OnInspectorGUI();
+                }
+            }
+
+            base.OnGUI(searchContext);
+        }
+
+        void DrawAssetSelection()
+        {
+            using (new EditorGUILayout.HorizontalScope())
+            {
+                var newSettings = (TGlobalSettings)EditorGUILayout.ObjectField(renderPipelineSettings, typeof(TGlobalSettings), false);
+
+                if (renderPipelineSettings != newSettings)
+                {
+                    if (newSettings != null)
+                        GraphicsSettings.RegisterRenderPipelineSettings<TRenderPipeline>(newSettings);
+                    else
+                    {
+                        if (EditorUtility.DisplayDialog($"Invalid {ObjectNames.NicifyVariableName(typeof(TGlobalSettings).Name)}", Styles.settingNullRPSettings, "Yes", "No"))
```

Comment:
It will be:

```
Invalid Universal Render Pipeline Global Settings (title)

Are you sure you want to unregister the Render Pipeline Settings? There might be issues on rendering.(settingNullRPSettings)

[Yes] [No]
```

Reply:
I think the pop up must be in as we need to make sure that the user specifies a valid global settings asset.

Reply:
Oh sorry, I misunderstood the dialog message. That message is much better, so it's fine.